gettsim/transfers/wohngeld.py

```diff
@@ -446,32 +391,80 @@ def wohngeld_miete_m_hh_ab_2021(
+    # introduced yet. For this time frame, the respective parameter is
+    # not part of wohngeld_params and heating allowance is set to 0.
+    if "heizkostenentlastung_m" in wohngeld_params:
+        max_def_hh_größe_heating = max(
+            i for i in wohngeld_params["heizkostenentlastung_m"] if isinstance(i, int)
+        )
+        if haushaltsgröße_hh <= max_def_hh_größe_heating:
+            heating_allowance_m = wohngeld_params["heizkostenentlastung_m"][
+                haushaltsgröße_hh
+            ]
+        else:
+            heating_allowance_m = wohngeld_params["heizkostenentlastung_m"][
+                max_def_hh_größe_heating
+            ] + (haushaltsgröße_hh - max_def_hh_größe_heating) * (
+                wohngeld_params["heizkostenentlastung_m"]["jede_weitere_person"]
+            )
     else:
-        heating_allowance = wohngeld_params["heizkosten_zuschuss"][
-            max_def_hh_größe_heating
-        ] + (haushaltsgröße_hh - max_def_hh_größe_heating) * (
-            wohngeld_params["heizkosten_zuschuss"]["jede_weitere_person"]
+        heating_allowance_m = 0
+
+    # Calc heating cost component. Before 2023, heating cost component was not
+    # introduced yet. For this time frame, the respective parameter is not part
+    # of wohngeld_params and heating cost component is set to 0.
+    if "dauerhafte_heizkostenkomponente_m" in wohngeld_params:
+        max_def_hh_größe_heating = max(
+            i
+            for i in wohngeld_params["dauerhafte_heizkostenkomponente_m"]
+            if isinstance(i, int)
         )
+        if haushaltsgröße_hh <= max_def_hh_größe_heating:
+            heating_component_m = wohngeld_params["dauerhafte_heizkostenkomponente_m"][
+                haushaltsgröße_hh
+            ]
+        else:
+            heating_component_m = wohngeld_params["dauerhafte_heizkostenkomponente_m"][
+                max_def_hh_größe_heating
+            ] + (haushaltsgröße_hh - max_def_hh_größe_heating) * (
+                wohngeld_params["dauerhafte_heizkostenkomponente_m"][
+                    "jede_weitere_person"
+                ]
+            )
+    else:
+        heating_component_m = 0
+
+    # Calc climate component. Before 2023, climate component was not
+    # introduced yet. For this time frame, the respective parameter is not
+    # part of wohngeld_params and climate component is set to 0.
+    if "klimakomponente_m" in wohngeld_params:
+        max_def_hh_größe_heating = max(
+            i for i in wohngeld_params["klimakomponente_m"] if isinstance(i, int)
+        )
+        if haushaltsgröße_hh <= max_def_hh_größe_heating:
+            climate_component_m = wohngeld_params["klimakomponente_m"][
+                haushaltsgröße_hh
+            ]
+        else:
+            climate_component_m = wohngeld_params["klimakomponente_m"][
+                max_def_hh_größe_heating
+            ] + (haushaltsgröße_hh - max_def_hh_größe_heating) * (
```

Comment:
Sorry, I've noticed one more thing - for households over 12 people we at ifo follow 
the Verwaltungsvorschriften from 2017 where in 19.31 Nr. 4 it says:

> Bei 14 zu berücksichtigenden Haushaltsmitgliedern ist nach § 19 Absatz 3 WoGG zu dem ermittelten Wohngeldbetrag für zwölf zu berücksichtigende Haushaltsmitglieder ein Betrag von zweimal 47 Euro hinzuzurechnen.

And in their example calculation they proceed the same for the old Heizkostenzuschuss. 

So for households that are bigger than 12 we just calculate ``dauerhafte_heizkostenkomponente_m``, ``klimakomponente_m`` and the heizkostenzuschuss for 12 people only and then add the ``bonus_12_mehr``. If I understand the code correctly you add on the amount for "jede_weitere_person" for everyone. 

We're not 100% sure this is correct, but it seems the most in line with the Verwaltungsvorschrift. It's hard to verify because all the official online calculators don't go above 12 people unfortunately. Feel free to disregard this if you disagree.

Reply:
Okay, thank you! I have incorporated it now. Could you have a final look, whether I understood your remark correctly?